providers/providers.go

```diff
@@ -72,6 +72,15 @@ func RegisterDomainServiceProviderType(name string, fns DspFuncs, pm ...Provider
 	unwrapProviderCapabilities(name, pm)
 }
 
+var ProviderMaintainers = map[string]string{}
+
+func RegisterMaintainer(
```

Comment:
What happens if a provider forgets to call this function?  We're back in the same situation where OWNERS gets out of date.

Instead of a function, how about adding the gitHubUsername as a parameter to `providers.RegisterRegistrarType` and `providers.RegisterDomainServiceProviderType`?

Benefits:
* An author can't forget to add it.
* "Volunteer needed" could be represented by `nil` or `""` (for strings, they are equiv).  `generateOwnersFile()` can generate the string "VOLUNTEER NEEDED" if the field is "".

Reply:
> Instead of a function, how about adding the gitHubUsername as a parameter to `providers.RegisterRegistrarType` and `providers.RegisterDomainServiceProviderType`?

This was my original implementation. However, this also has disadvantages. You must maintain double administration for each provider that has a `RegisterRegistrarType()` and a `RegisterDomainServiceProviderType()`.

**Separate function `RegisterMaintainer()`**

```diff
providers.RegisterDomainServiceProviderType(providerName, fns, features)
providers.RegisterRegistrarType(providerName, newRoute53Reg)
+providers.RegisterMaintainer(providerName, providerMaintainer)
```

**Parameter to `providers.RegisterRegistrarType` and `providers.RegisterDomainServiceProviderType`**

```diff
-providers.RegisterDomainServiceProviderType(providerName, fns, features)
+providers.RegisterDomainServiceProviderType(providerName, providerMaintainer, fns, features)
-providers.RegisterRegistrarType(providerName, newRoute53Reg)
+providers.RegisterRegistrarType(providerName, providerMaintainer, newRoute53Reg)
```

Personally, I find the separate `RegisterMaintainer()` function clearer/more organized.

Reply:
Good point!

Ok, here's a more radical solution 

An more "modern golang" pattern is to pass a struct with a number of options:

```
providers.Register(providers.RegisterOptions{
     Name: "POWERDNS",
     Maintainer: "@jpbede",
     Features: features,
     RegistrarInitializer: newRoute53Reg,
     DnsInitializer: newDSP,
     AuditRecords: AuditRecords,
})
```

If something is not a registrar, leave out RegistrarInitializer; if it is not a dns provider, leave out DnsInitializer.

This could be in a separate PR. It seems unrelated to generating the OWNERS file.